### PR TITLE
[API] Better configuration for allowed paths in the `/files` API

### PR DESCRIPTION
**Pull request by theSaarco**
For security reasons, we have limited the URLs accessible through the `/files` API such that only `v3io://` paths are allowed by default. There is one additional tweak, where the parameters `httpdb.data_volume` and `httpdb.real_path` can be specified. These were intended such that if the path is prefixed with `data_volume`, it will be replaced with the `real_path`. As a side-effect, the `real_path` is allowed for the `/files` API, so specifying something like `gcs` there will allow access to paths beginning with `gcs` (even if `data_volume` is omitted).

This is not flexible enough, as users may want to specify paths to specific buckets, for example. The current mechanism does not allow that since only a single prefix can be provided.

The PR adds a new parameter - `httpdb.allowed_file_paths` which contains a comma separated prefixes of paths that will be allowed. For example, to access the buckets `bucket1` and `another_bucket` on gcs, the following value can be provided: `"gcs://bucket1,gcs://another_bucket"`. Values provided must contain schemas (must have `://` in them) to prevent abuse such as specifying `/` to access local files. By default this value is empty, and it should be modified with care for specific usages. 
It is fully backwards compatible since the `real_path` is still considered an allowed path, as well as `v3io://`.

### CODE REVIEW

File: mlrun/api/api/utils.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def project_logs_path(project) -> Path:
 
 
 def get_obj_path(schema, path, user=""):
```
Comment (Tankilevitch) [inline mlrun/api/api/utils.py:50]: I know you only added the last section but this function seems not documented at all, can you please add a docstring or comments for why we have all those conditions, e.g `/User` and where/when it comes from etc. Thanks

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, added a comment.

File: mlrun/api/api/utils.py [hunk 2]
```diff
@@ -59,19 +60,29 @@ def get_obj_path(schema, path, user=""):
...
-    ):
+
+    # Check if path is allowed - v3io:// is always allowed, and also the real_path parameter if specified.
+    # We never allow local files in the allowed paths list. Allowed paths must contain a schema (://)
+    allowed_file_paths = config.httpdb.allowed_file_paths
+    allowed_paths_list = (
+        [path.strip() for path in allowed_file_paths.split(",") if "://" in path]
+        if allowed_file_paths
+        else []
+    )
```
Comment (quaark) [inline mlrun/api/api/utils.py:80]:
```suggestion
    allowed_file_paths = config.httpdb.allowed_file_paths or ""
    allowed_paths_list = [path.strip() for path in allowed_file_paths.split(",") if "://" in path]
```

Comment (quaark) [reply mlrun/api/api/utils.py:80]: Just a suggestion to make the code more concise

**Review by quaark — APPROVED** [review]
LGTM, a small suggestion

File: mlrun/api/api/utils.py (hunk 2 above)
Comment (theSaarco) [reply mlrun/api/api/utils.py:80]: Issue with this is that `split` *with delimiter* on an empty string will return a non-empty array (`['']`). Then checking `startswith` on the empty string will not give us correct results.
It is an annoying issue with the fact that `split` implements two different mechanisms when a delimiter is provided and when it is not - [see here](https://stackoverflow.com/questions/16645083/when-splitting-an-empty-string-in-python-why-does-split-return-an-empty-list#:~:text=In%20the%20case%20of%20splitting,produce%20the%20first%20empty%20field.)

File: mlrun/api/api/utils.py (hunk 1 above)
Comment (theSaarco) [reply mlrun/api/api/utils.py:50]: Done.

File: mlrun/api/api/utils.py (hunk 2 above)
Comment (theSaarco) [reply mlrun/api/api/utils.py]: After discussing this, made the change.

**Review by Tankilevitch — APPROVED** [review]
Great description and comments, Approved.

**Pull request merged**